src/grpc_frame.rs

```diff
@@ -187,15 +187,22 @@ impl Stream for GrpcFrameFromHttpFramesStreamResponse {
                 };
                 match headers.content {
                     HttpStreamPartContent::Headers(headers) => {
-                        let status = slice_get_header(&headers, ":status");
-                        if status != Some("200") {
-                            self.error = Some(stream::once(Err(if let Some(message) = slice_get_header(&headers, HEADER_GRPC_MESSAGE) {
-                                GrpcError::GrpcMessage(GrpcMessageError { grpc_message: message.to_owned() })
-                            } else {
-                                GrpcError::Other("not 200")
-                            })));
+                        if slice_get_header(&headers, ":status") != Some("200") {
+                            self.error = Some(stream::once(Err(GrpcError::Other("not 200"))));
                             continue;
                         }
+
+                        // Check gRPC status code and message
+                        // TODO: a more detailed error message.
+                        if let Some(code) = slice_get_header(&headers, HEADER_GRPC_STATUS) {
+                            if code != OK.to_string() {
```

Comment:
I think `to_string` is fine, it will not cause performance issues. Besides, codes are digits not `'static str`.

See more: https://github.com/grpc/grpc-go/blob/master/codes/codes.go#L39

>// A Code is an unsigned 32-bit error code as defined in the gRPC spec.
type Code uint32